### PR TITLE
feat: add information for about libraries support

### PR DESCRIPTION
**Pull request by ShaunPlummer**
Include additional string resources for use by AboutLibraries to display
information about this library when included as a project dependency.

This change follows the guidelines [here](https://github.com/mikepenz/AboutLibraries/wiki/HOWTODEV:-Include-in-your-library) and has been tested by including the project as a submodule in AboutLibraries.

image

### CODE REVIEW

**Comment by Chesire** [comment]
Build failure looks to be
`Ensure required Android SDK components
Retrying...
Failed to ensure android components, error: output: > Configure project :lifecyklelog
fatal: No names found, cannot describe anything.
FAILURE: Build failed with an exception.
* Where:
Build file '/bitrise/src/lifecyklelog/build.gradle' line: 11
* What went wrong:
A problem occurred evaluating project ':lifecyklelog'.
> Process 'command 'git'' finished with non-zero exit value 128
`

File: lifecyklelog/src/main/res/values/library_info.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="define_Lifecyklelog" translatable="false" ></string>
+    <string name="library_Lifecyklelog_author" translatable="false" >Chesire</string>
+    <string name="library_Lifecyklelog_authorWebsite" translatable="false" >https://github.com/Chesire</string>
+    <string name="library_Lifecyklelog_libraryName" translatable="false" >Lifecyklelog</string>
```
Comment (Chesire) [inline lifecyklelog/src/main/res/values/library_info.xml]: This should be `LifecykleLog` to match Library name

File: lifecyklelog/build.gradle [hunk 2]
```diff
@@ -5,9 +5,26 @@ apply from: 'bintray.gradle'
...
+    def stdout
+    stdout = new ByteArrayOutputStream()
+    exec {
+        commandLine 'git', 'describe', '--tags'
+        standardOutput = stdout
+    }
+    def longTag = stdout.toString().trim()
+
+    return longTag.substring(longTag.indexOf('/') + 1, longTag.indexOf('-'))
+
```
Comment (Chesire) [inline lifecyklelog/build.gradle]: There should be no blank lines at end of method

**Comment by ShaunPlummer** [comment]
Weird, project evaluation is passing for me locally but the 'build' task still fails because of a no class def error when running the unit tests. I've made a speculative change and moved the property declaration to the project root.

**Comment by ShaunPlummer** [comment]
Would you mind sharing the bitrise yaml?

**Comment by Chesire** [comment]
Bitrise yml
```---
format_version: '7'
default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
project_type: android
trigger_map:
- pull_request_source_branch: "*"
  workflow: primary
workflows:
  primary:
    steps:
    - activate-ssh-key@4.0.3:
        run_if: '{{getenv "SSH_RSA_PRIVATE_KEY" | ne ""}}'
    - git-clone@4.0.17: {}
    - cache-pull@2.1.0: {}
    - install-missing-android-tools@2.3.7:
        inputs:
        - gradlew_path: "$PROJECT_LOCATION/gradlew"
    - android-lint@0.9.6:
        inputs:
        - project_location: "$PROJECT_LOCATION"
        - module: "$MODULE"
        - variant: "$VARIANT"
    - android-unit-test@0.11.1:
        inputs:
        - project_location: "$PROJECT_LOCATION"
        - module: "$MODULE"
        - variant: "$VARIANT"
    - gradle-runner@1.9.0:
        title: jacocoTestReport
        inputs:
        - gradle_task: "$MODULE:jacocoTestReport"
    - android-detekt@0.0.2: {}
    - gradle-runner@1.9.0:
        title: ktlint
        inputs:
        - gradle_task: ktlint
    - codecov@1.1.5:
        inputs:
        - CODECOV_TOKEN: "$CODECOV_TOKEN"
    - deploy-to-bitrise-io@1.6.1: {}
    - cache-push@2.2.0: {}
    - slack@3.1.3:
        inputs:
        - channel: "#lifecyklelog"
        - fields: |-
            App|${BITRISE_APP_TITLE}
            Branch|${BITRISE_GIT_BRANCH}
            Workflow|${BITRISE_TRIGGERED_WORKFLOW_ID}
            Build|#${BITRISE_BUILD_NUMBER}
        - channel_on_error: "#lifecyklelog"
        - api_token: "$SLACK_BOT_TOKEN"
app:
  envs:
  - opts:
      is_expand: false
    PROJECT_LOCATION: "."
  - opts:
      is_expand: false
    MODULE: lifecyklelog
  - opts:
      is_expand: false
    VARIANT: ''
  - opts:
      is_expand: false
    GRADLEW_PATH: gradlew
```
Doesn't look like its doing anything special when its trying to build

**Comment by Chesire** [comment]
Looks to be because it shallow clones the repository, the `git describe --tags` invocation is failing. Try the below block, I think that will work
```
ext.getVersionTag = {
    def stdout = new ByteArrayOutputStream()
    exec {
        commandLine 'git', 'describe', '--tags', '--always'
        standardOutput = stdout
    }
    def longTag = stdout.toString().trim()

    return longTag.substring(longTag.indexOf('/') + 1, longTag.indexOf('-'))
}

**Comment by ShaunPlummer** [comment]
I see the tags aren't being cloned along with the repo, so the following line returns the commit hash, not the tag. `commandLine 'git', 'describe', '--tags', '--always'`

The recommendation from Bitrise is to add a setup script running something like `git fetch <remote> 'refs/tags/*:refs/tags/*'` to fetch only tags without associated commits

Currently bitrise is performing:
```
git "init"
git "remote" "add" "origin" "git@github.com:ShaunPlummer/LifecykleLog.git"
git "fetch" "origin" "testing"
git "checkout" "testing"
git "merge" "origin/testing"
git "submodule" "update" "--init" "--recursive"
```

**Comment by ShaunPlummer** [comment]
I've tested this added to the bitrise.yml and it appears to be working

```
    - script@1.1.5:
        inputs:
        - content: |-
            #!/usr/bin/env bash
            # fail if any commands fails
            set -e
            # debug log
            set -x

            git fetch origin 'refs/tags/*:refs/tags/*'
```

**Comment by Chesire** [comment]
Looks like that added script fixed it 👍 merged in

**Pull request merged**